src/hotspot/os/posix/os_posix.hpp

```diff
@@ -92,22 +92,7 @@ class Posix {
   static address ucontext_get_pc(const ucontext_t* ctx);
   static void    ucontext_set_pc(ucontext_t* ctx, address pc);
 
-#ifdef SUPPORTS_CLOCK_MONOTONIC
-
-private:
-  static bool _supports_monotonic_clock;
-  // These need to be members so we can access them from inline functions
-  static int (*_clock_gettime)(clockid_t, struct timespec *);
-  static int (*_clock_getres)(clockid_t, struct timespec *);
-public:
   static bool supports_monotonic_clock();
```

Comment:
Why do we need this API at all now?

thread.cpp uses it here:

      assert(!os::supports_monotonic_clock(),
             "unexpected time moving backwards detected in JavaThread::sleep()");

so we can just remove this usage?